### ISSUE TITLE
dataFrom does not function with permitted-key-name annotation

### ISSUE THREAD

**Issue by Strange-Account** [godaddy/kubernetes-external-secrets#291]
Hi,

Version: 3.1.0
Install: by Helm Chart
Environment: GKE 1.14 cluster

this doesn't function when the namespace has the permitted-key-name annotation:

```
apiVersion: v1
kind: Namespace
metadata:
  name: ns-a
  annotations:
    externalsecrets.kubernetes-client.io/permitted-key-name: "/dev/poc/a/.*"

apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: credentials
spec:
  backendType: secretsManager
  dataFrom:
    - /dev/poc/a/credentials
```

The error message is: "ERROR, Cannot read property 'forEach' of undefined"

When I remove the annotation or add a data spec to the external secret it is okay, eg.
```
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: credentials
spec:
  backendType: secretsManager
  data:
    - key: /dev/poc/a/credentials
      name: password
      property: password
  dataFrom:
    - /dev/poc/a/credentials
```

**Comment by Strange-Account** [comment]
It gets worse:

When i use:

apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
name: credentials
spec:
backendType: secretsManager
data:
- key: /dev/poc/a/credentials
name: password
property: password
dataFrom:
- /dev/poc/b/credentials

then the dataFrom part is not checked against the naming convention and is merged.

**Comment by Flydiverny** [comment]
Fixed in #292
